[효오, 비모] 댓글 CRUD

hyojaekim opened the pull request:
테스트는 계속해서 추가하고 있습니다 ㅠ

hyojaekim committed:
chore: dependency 추가

hyojaekim committed:
feat: Comment 작성 기능 추가

Comment 작성 API 추가

hyojaekim committed:
feat: Comment 수정 기능 추가

Comment 수정 API 추가

hyojaekim committed:
feat: Comment 삭제 기능 추가

Comment 삭제 API 추가

hyojaekim committed:
chore: 불필요한 설정 제거

hyojaekim committed:
feat: Comment 작성 기능 추가

1. 영상 페이지에 댓글 HTML 합치기
2. 댓글 작성 관련 Javascript 코드 추가
3. 댓글 작성 관련 CSS 추가

hyojaekim committed:
refactor: Comment 예외 처리 가이드 추가

hyojaekim committed:
feat: 없는 Comment에 대한 Exception 추가

hyojaekim committed:
refactor: Comment 생성자 Lombok 적용 및 Column 속성 추가

hyojaekim committed:
refactor: Comment 연관 관계 가이드 추가

hyojaekim committed:
test: 댓글 생성 테스트 추가

hyojaekim committed:
test: Comment 수정 테스트 추가

hyojaekim committed:
test: 댓글 삭제 테스트 추가

hyojaekim committed:
refactor: Comment Test 가이드 추가

hyojaekim committed:
test: 존재하지 않는 comment update 테스트 추가

hyojaekim committed:
test: 존재하지 않는 Comment 삭제 테스트 추가

hyojaekim committed:
style: 컨벤션에 맞춰 코드 포맷 수정

hyojaekim committed:
test: Comment Service 댓글 생성 테스트 추가

hyojaekim committed:
refactor: Comment EqualsAndHashCode 추가

hyojaekim committed:
test: Comment Service 댓글 수정 테스트 추가

hyojaekim committed:
refactor: CommentResponse 생성자 제거

hyojaekim committed:
refactor: 패키지 이동

hyojaekim committed:
test: Comment Service 댓글 삭제 테스트 추가

1. 댓글이 없는 경우
2. 댓글이 있는 경우

hyojaekim committed:
test: Comment Service 존재하지 않은 댓글 수정 테스트 추가

hyojaekim committed:
refactor: Comment Service 예외 처리 가이드 추가

hyojaekim committed:
refactor: 구독 버튼 테두리 제거

hyojaekim committed:
style: 패키지 이동

hyojaekim committed:
style: Comment 관련 Dto 이름 수정

hyojaekim committed:
style: Comment 조회 기능 가이드 추가

yjll1019 reviewed (approved):
중간 중간 주석 처리로 아직 구현되지 않은 부분을 적어주셔서 쉽게 이해할 수 있었습니다 👍

SEOKHYOENCHOI committed:
feat: comment front 코드 추가

1. 수정
2. 삭제

hyojaekim committed:
feat: Reply Entity 생성

1. 업데이트 로직 추가

hyojaekim committed:
feat: Reply 생성 및 수정 기능 추가

hyojaekim committed:
feat: Reply 예외 처리

1. 존재하지 않는 답글인 경우
2. ExceptionHandler 추가

hyojaekim committed:
test: Reply 생성 및 수정 테스트 추가

- 상속을 이용한 테스트 중복 제거

hyojaekim committed:
refactor: Comment 찾는 메소드 추가

hyojaekim committed:
refactor: Reply 메서드 분리

- 댓글이랑 매치되지 않는 경우 예외를 던지는 로직

hyojaekim committed:
feat: Reply 삭제 기능 추가

- 중복 Path 수정
- 예외 처리

hyojaekim committed:
test: Reply 삭제 테스트 추가

1. 답글을 삭제하는데 성공하는 테스트
2. 다른 댓글의 답글을 삭제하는 경우 실패하는 테스트
3. 존재하지 않는 답글을 삭제하는 경우 실패하는 테스트

SEOKHYOENCHOI committed:
feat: comment front 코드 추가

프로필 사진 및 버튼 이벤트 추가

SEOKHYOENCHOI committed:
refactor: 공통 코드 분리

hyojaekim committed:
test: Rest Assured 적용

backlo commented on src/main/java/com/wootube/ioi/domain/model/Comment.java:
```diff
@@ -0,0 +1,53 @@
+//    @JoinColumn
+//    @OnDelete(action = OnDeleteAction.CASCADE)
+//    private Video video;
+
```
주석처리 부분 빼주세요~ 😄

SEOKHYOENCHOI replied:
Video와 합칠 때 적용하기 위해서 주석처리 했어요!
합치면서 주석 제거할게요

backlo commented on src/main/java/com/wootube/ioi/domain/model/Comment.java:
```diff
@@ -0,0 +1,53 @@
+@Getter
+@Entity
+@NoArgsConstructor
+@EqualsAndHashCode(of = "id")
```
of 사용 굳! 👍

backlo commented on src/main/java/com/wootube/ioi/service/CommentService.java:
```diff
@@ -0,0 +1,59 @@
+//    public List<CommentResponseDto> findByVideoId(Long videoId) {
+//        return Collections.unmodifiableList(commentRepository.findAllByVideoId(videoId));
+//    }
+
```
여기도 주석 빼주세요~~

backlo commented on src/main/java/com/wootube/ioi/service/CommentService.java:
```diff
@@ -0,0 +1,59 @@
+        comment.update(commentRequestDto.getContents());
+        return new CommentResponseDto(comment.getId(),
+                comment.getContents(),
+                comment.getUpdateTime());
```
Builder 패턴 쓰시는게 어떨까요??

backlo commented on src/main/java/com/wootube/ioi/service/dto/CommentRequestDto.java:
```diff
@@ -0,0 +1,12 @@
+import lombok.NoArgsConstructor;
+
+@Getter
+@AllArgsConstructor
```
lombok 사용시 NoArgs빼고 사용을 지양하라고 되어있습니다. AllArgs말고 빌더패턴을 사용해 보는게 어떨까요????
참고자료 입니다 😄 
https://kwonnam.pe.kr/wiki/java/lombok/pitfall

SEOKHYOENCHOI replied:
Model <-> Dto 변환을 위해서 사용한 Annotation인데, 글을 읽어보니 문제가 생길 수도 있을 것 같네요 👍 
빌더패턴이나 Assembler를 사용하면 각 Dto, Model 마다 계속 변환 메서드를 작성해야돼서 불편함이 있을 것 같아요.
ModelMapper를 적용해볼게요.

backlo reviewed (changes requested):
구현 잘해 주셨네요! 💯 
몇개 없지만 피드백 드립니다~~

Commit:
Conflict 해결

Commit:
fix: 테스트 어노테이션 추가
